x-pack/plugin/shutdown/src/internalClusterTest/java/org/elasticsearch/xpack/shutdown/NodeShutdownShardsIT.java

```diff
@@ -391,6 +369,33 @@ public Settings onNodeStopped(String nodeName) throws Exception {
         });
 
         ensureGreen("myindex");
+        assertion.verify();
+    }
+
+    public void testAutoExpandDuringReplace() throws Exception {
+
+        var node1 = internalCluster().startNode();
+        var node2 = internalCluster().startNode();
+
+        createIndex("index", indexSettings(1, 0).put("index.auto_expand_replicas", "0-1").build());
+        ensureGreen("index");
+
+        var assertion = new BackgroundAssertion(() -> assertIndexSetting("index", "index.number_of_replicas", "1"));
```

Comment:
Can we instead just validate in between each step. Perhaps also validate that the index is green between every step.